Objects/unicodeobject.c

```diff
@@ -700,15 +700,15 @@ _PyUnicode_CheckConsistency(PyObject *op, int check_content)
             CHECK(ascii->state.compact == 0);
             CHECK(data != NULL);
             if (ascii->state.ascii) {
-                CHECK(compact->utf8 == data);
+                CHECK(_PyUnicode_UTF8(op) == data);
                 CHECK(compact->utf8_length == ascii->length);
             }
             else {
-                CHECK(compact->utf8 != data);
+                CHECK(_PyUnicode_UTF8(op) != data);
             }
         }
 
-        if (compact->utf8 == NULL)
+        if (_PyUnicode_UTF8(op) == NULL)
```

Comment:
This check is not safe in the free threading build: if `utf8` is not NULL, then the length is valid, but the opposite condition doesn't tell you anything.

I think we should just `#ifndef Py_GIL_DISABLED` around this.